targets/TARGET_Cypress/TARGET_PSOC6/cy_sleep_api.c

```diff
@@ -26,11 +26,16 @@
 
 void hal_sleep(void)
 {
+    // Noop, if the idle mode is active
+#if !defined(CY_CFG_PWR_SYS_IDLE_MODE) || (CY_CFG_PWR_SYS_IDLE_MODE != CY_CFG_PWR_MODE_ACTIVE)
     cyhal_syspm_sleep();
+#endif
```

Comment:
Question: As the macro is Cypress-specific, maybe it makes more sense to have that defined in `cyhal_syspm_sleep()` (Cypress SDK code) instead of the Mbed OS HAL implementation?

Reply:
@ARMmbed/team-cypress Or, could you confirm if the macro is still being used by Cypress's debug tool as of today?

Reply:
That's a good suggestion. 
@kyle-cypress please review this proposal and let us know if you'd rather change your SDK.

Reply:
I don't think we should change cyhal_syspm_sleep. The intent of CY_CFG_PWR_SYS_IDLE_MODE  is to control how the RTOS idle task behaves, not to completely disable the sleep driver (even if the two wind up being very similar in practice).

Reply:
The current proposal is good and easy to follow so I would go with it.

Reply:
> The intent of CY_CFG_PWR_SYS_IDLE_MODE is to control how the RTOS idle task behaves,

Well, as explained in the description, the idle overriding is a feature we provide to the application. Overriding it by drivers/BSPs would create a conflict if the application uses it.